test/util/kubeflowjob_utils.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+)
+
+const (
+	priorityClassName = "test-priority-class"
+	priorityValue     = 10
+	jobQueueName      = "test-queue"
+)
+
+func CheckCreatedWorkload(ctx context.Context, client client.Client, wlLookupKey types.NamespacedName, createdJob metav1.Object) *kueue.Workload {
+	createdWorkload := &kueue.Workload{}
+	gomega.Eventually(func() error {
+		return client.Get(ctx, wlLookupKey, createdWorkload)
+	}, Timeout, Interval).Should(gomega.Succeed())
+	gomega.Expect(createdWorkload.Spec.QueueName).Should(gomega.Equal(""), "The Workload shouldn't have .spec.queueName set")
+	gomega.Expect(metav1.IsControlledBy(createdWorkload, createdJob)).To(gomega.BeTrue(), "The Workload should be owned by the Job")
+
+	ginkgo.By("checking the workload is created with priority and priorityName")
+	gomega.Expect(createdWorkload.Spec.PriorityClassName).Should(gomega.Equal(priorityClassName))
+	gomega.Expect(*createdWorkload.Spec.Priority).Should(gomega.Equal(int32(priorityValue)))
+
+	return createdWorkload
+}
+
+func VerifyWorkloadQueueName(ctx context.Context, client client.Client, createdWorkload *kueue.Workload, wlLookupKey types.NamespacedName) {
+	gomega.Eventually(func() bool {
+		if err := client.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+			return false
+		}
+		return createdWorkload.Spec.QueueName == jobQueueName
+	}, Timeout, Interval).Should(gomega.BeTrue())
+}
+
+func CreateAdmission(createdWorkload *kueue.Workload, clusterQueueName string, masterCpu, workerCpu kueue.ResourceFlavorReference) *kueue.Admission {
```

Comment:
I don't have a strong view, but I'm not convinced about extracting the admission configuration as a function. 

The reason is that when reading the test, it is not clear that the admission creates a "worker" and "master" pod sets. So it requires a jump between files to match the configuration with asserts later. 

I think I would like to extract functions for which the name is descriptive enough that you can get what they do without a jump, for example `AwaitWorkloadQueueName`.

Reply:
Configuration is the same for each test, so I think it make sense to define a separate function for that. It makes the function smaller and I think it improves readability

Reply:
I see it helps to commonize, which is a gain, but it splits the bits which need to match between two files. Maybe  https://github.com/kubernetes-sigs/kueue/pull/1191/files#r1353109319 would help to handle both aspects.